src/vs/workbench/api/common/extHostTerminalService.ts

```diff
@@ -248,6 +248,9 @@ export class ExtHostPseudoterminal implements ITerminalChildProcess {
 		if (this._pty.onDidOverrideDimensions) {
 			this._pty.onDidOverrideDimensions(e => this._onProcessOverrideDimensions.fire(e ? { cols: e.columns, rows: e.rows } : e));
 		}
+		if (this._pty.onDidChangeName) {
+			this._pty.onDidChangeName(title => this._onProcessTitleChanged.fire(title));
```

Comment:
This event already existed but it was never being fired. 

https://github.com/microsoft/vscode/blob/81ae94a9dc82197829717915db783e3b934db46a/src/vs/workbench/api/common/extHostTerminalService.ts#L477-L481